### ISSUE TITLE
Collect crashes when user minimizes app during deleting blank forms when Don't keep activity is turned on

### ISSUE THREAD

**Issue by mmarciniak90** [getodk/collect#4303]
#### Software and hardware versions 
Collect v1.29 Beta 1, master
Verified on Android 7.0, 8.1, 10.0

#### Problem description
Collect crashes when user minimizes app during deleting blank forms when Don't keep activity is turned on

#### Steps to reproduce the problem
1. Don't keep activities is turned on
2. User goes to Delete Saved Forms -> **Blank Forms**
3. User selects all forms
4. User clicks on Delete Selected -> dialog with progress bar displays
5. User minimizes an app
6. User goes back to Collect
7. Collect crashes [crash.txt](https://github.com/getodk/collect/files/5689520/crash.txt.txt)

#### Expected behavior
Dialog is restored

#### Other information 
Related to #3843

@getodk-bot label "regression"

**Comment by lognaturel** [comment]
I put this in v1.29 so we don't forget about it but it's definitely not a must have.

**Comment by grzesiek2010** [comment]
I agree that this is not important and we can skip it for now. I created a pr https://github.com/getodk/collect/pull/4307 which doesn't solve this issue but at least avoid crashing. I think we should merge it without testing and work on better user experience (recreating the dialog) later.

**Comment by seadowg** [comment]
A note to myself here: I do wonder if we can find a way to get rid of `executePendingTransactions` as it feels really dodgy (pretty sure I added though). I'm sure at the time we did a bunch of work to see if there was a better way to make sure we don't end up with multiple dialogs, but I'd like to have another look at that.
